java-checks/src/main/java/org/sonar/java/checks/helpers/QuickFixHelper.java

```diff
@@ -118,4 +136,99 @@ public static String contentForTree(Tree tree, JavaFileScannerContext context) {
 
     return sb.toString();
   }
+
+  public static Optional<JavaTextEdit> newImportEdit(String requiredType, Tree base) {
+    if (!requiresImportOf(requiredType, base)) {
+      return Optional.empty();
+    }
+
+    CompilationUnitTree cut = ((JavaTree) base).root();
+    List<ImportClauseTree> imports = cut.imports();
+
+    if (imports.isEmpty()) {
+      PackageDeclarationTree packageDeclaration = cut.packageDeclaration();
+      if (packageDeclaration != null) {
+        // could be a normal compilation unit or a package-info file: 2 lines after the package declaration
+        return Optional.of(JavaTextEdit.insertAfterTree(packageDeclaration, String.format("%n%nimport %s;", requiredType)));
+      }
+      // default package
+      List<Tree> types = cut.types();
+      if (!types.isEmpty()) {
+        // two lines before the first type
+        return Optional.of(JavaTextEdit.insertBeforeTree(types.get(0), String.format("import %s;%n%n", requiredType)));
+      }
+      // no package declaration and no type? Should be impossible or an empty file
+      return Optional.empty();
+    }
+
+    List<ImportTree> nonStaticImports = imports
+      .stream()
+      .filter(importClauseTree -> importClauseTree.is(Tree.Kind.IMPORT))
+      .map(ImportTree.class::cast)
+      .filter(importTree -> !importTree.isStatic())
+      .sorted(IMPORT_TREE_COMPARATOR)
```

Comment:
I'm wondering if we should try to find something smarter than sorting the whole list of imports for every call to this method, "just" to find the closest matching import or if it would be overengineering because it will be called only on rare occasions when we have an issue with a quick fix?

Reply:
to be honest, I feel it's OK to do it every time. I also don't see any "smart" way to do it only a few times.